packages/eas-cli/src/project/publish.ts

```diff
@@ -0,0 +1,255 @@
+import { Platform } from '@expo/config';
+import JsonFile from '@expo/json-file';
+import crypto from 'crypto';
+import fs from 'fs';
+import { uniqBy } from 'lodash';
+import mime from 'mime';
+import path from 'path';
+
+import { AssetMetadataStatus, PartialManifestAsset } from '../graphql/generated';
+import { PublishMutation } from '../graphql/mutations/PublishMutation';
+import { PublishQuery } from '../graphql/queries/PublishQuery';
+import { PresignedPost, uploadWithPresignedPostAsync } from '../uploads';
+
+export const TIMEOUT_LIMIT = 60_000; // 1 minute
+const STORAGE_BUCKET = getStorageBucket();
+export const Platforms: PublishPlatforms[] = ['android', 'ios']; // TODO-JJ allow users to specify this in app.js
+
+function getStorageBucket(): string {
+  if (process.env.NODE_ENV === 'test') {
+    return 'update-assets-testing';
+  } else if (process.env.EXPO_STAGING || process.env.EXPO_LOCAL) {
+    return 'update-assets-staging';
+  } else {
+    return 'update-assets-production';
+  }
+}
+
+export type PublishPlatforms = Extract<'android' | 'ios', Platform>;
```

Comment:
```suggestion
export type PublishPlatform = Extract<'android' | 'ios', Platform>;
```
(The prevailing convention is to use singular names for unions.)